applications/tari_merge_mining_proxy/src/block_template_data.rs

```diff
@@ -37,18 +60,43 @@ impl BlockTemplateRepository {
     }
 
     pub async fn get<T: AsRef<[u8]>>(&self, hash: T) -> Option<BlockTemplateData> {
+        trace!(
+            target: LOG_TARGET,
+            "Retrieving blocktemplate with merge mining hash: {:x?}",
+            hash.as_ref()
+        );
         let b = self.blocks.read().await;
-        b.get(hash.as_ref()).cloned()
+        match b.get(hash.as_ref()).cloned() {
+            Some(item) => Some(item.data),
+            None => None,
+        }
     }
 
     pub async fn save<T: AsRef<[u8]>>(&mut self, hash: T, block_template: BlockTemplateData) {
+        trace!(
+            target: LOG_TARGET,
+            "Saving blocktemplate with merge mining hash: {:x?}",
+            hash.as_ref()
```

Comment:
Still not printing as hex:
```
2021-01-13 11:52:16.538742100 [tari_mm_proxy::xmrig] TRACE Saving blocktemplate with merge mining hash: [1f, cc, c4, aa, c8, de, 45, 59, 6f, 6c, 33, 18, 2d, 5e, 16, 25, ab, 3, 52, 2a, cb, 4e, 73, 42, 5e, 83, 80, f1, 39, 47, 6a, 91] 
```

Reply:
Ah. Think we had a misunderstanding here, I though you wanted the bytes in the array displayed as hex instead of numeric (as in the above). I see you now mean the hex representation as a string. Will sort it out.